packages/block-editor/src/hooks/layout.js

```diff
@@ -413,7 +408,7 @@ export const withLayoutStyles = createHigherOrderComponent(
 						/>,
 						element
 					) }
-				<BlockListBlock { ...props } className={ className } />
+				<BlockListBlock { ...props } layoutClassNames={ className } />
```

Comment:
My original idea here was not to add a new prop to BlockListBlock and try to pass it down using context like we did but maybe we could try a dedicated filter for the inner blocks component instead of relying on `editor.BlockListBlock`.

Adding filters has its own drawbacks too so we must be careful regardless of the approach. Any approach requires a new API either a prop or a filter. I'd prefer an "internal/unstable" API.

Reply:
If we make it a filter for InnerBlocks, we'll have to also create a hook to add the layout classnames to blocks that don't use InnerBlocks, like Gallery and the Placeholder version of Post Content. The advantage of passing the classnames through BlockListBlock and into BlockEdit is that we can retrieve them directly from the Edit props in those cases. 

I guess we can make this an "unstable" prop, but we do need to start thinking of what shape the layout API should take when stabilised. Even with this prop being unstable, the layout classnames will be output as props to the Edit function, which kind of makes them public by default 😅

Reply:
> the Placeholder version of Post Content

I think these get the "layout" from their attributes anyway and they already know that they support "layout"

Reply:
They know their layout attributes, but we need to add the layout classnames manually because those blocks don't use InnerBlocks. We can't work out the classnames from the attributes because there's no way of knowing the `wp-container` class so there has to be a way to get the classnames in the blocks' edit function.